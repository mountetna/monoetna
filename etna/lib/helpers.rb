require_relative './etna/clients'
require_relative './etna/environment_scoped'

module WithEtnaClients
  def environment
    EtnaApp.instance.environment
  end

  def token
    if !ENV['TOKEN']
      puts "No environment variable TOKEN is set. Set your token with `export TOKEN=<your.janus.token>` before running Etna commands."
      exit
    end
    ENV['TOKEN']
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(token: token, host: EtnaApp.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(token: token, host: EtnaApp.instance.config(:metis, environment)[:host])
  end

  def janus_client
    @janus_client ||= Etna::Clients::Janus.new(token: token, host: EtnaApp.instance.config(:janus, environment)[:host])
  end
end

module WithLogger
  def logger
    EtnaApp.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
