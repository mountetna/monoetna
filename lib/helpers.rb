require_relative './etna/clients'
require_relative './etna/environment_scoped'

module WithEtnaClients
  def environment
    EtnaApp.instance.environment
  end

  def exit(status=true)
    WithEtnaClients.exit(status)
  end

  # Abstraction used to prevent accidental exist in specs.
  def self.exit(status)
    Kernel.exit(status)
  end

  def token(ignore_environment: false)
    unless ignore_environment
      if environment == :many
        raise "You have multiple environments configured, please specify your environment by adding --environment <staging|production|development>"
      elsif environment == :none
        raise "You do not have a successfully configured environment, please run #{program_name} config set https://polyphemus.ucsf.edu"
      end
    end

    env_token = ENV['TOKEN']
    if !env_token
      puts "No environment variable TOKEN is set.  You should set your token with `export TOKEN=<your.janus.token>` before running."
      redirect = EtnaApp.instance.config(:auth_redirect)

      if redirect.nil? && EtnaApp.instance.environment == :production
        redirect = 'https://janus.ucsf.edu/'
      end

      unless redirect.nil?
        puts "Open your browser to #{redirect} to complete login and copy your token."
      end

      exit
    end

    env_token
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:magma, environment) || {})
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:metis, environment) || {})
  end

  def janus_client
    @janus_client ||= Etna::Clients::Janus.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:janus, environment) || {})
  end

  def polyphemus_client
    @polyphemus_client ||= Etna::Clients::Polyphemus.new(
        token: token,
        ignore_ssl: EtnaApp.instance.config(:ignore_ssl),
        **EtnaApp.instance.config(:polyphemus, environment) || {})
  end
end

module WithLogger
  def logger
    EtnaApp.instance.logger
  end
end

module StrongConfirmation
  def confirm
    puts "Confirm Y/n:"
    input = STDIN.gets.chomp
    if input.downcase != "y"
      puts "Bailing..."
      return false
    end

    true
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
