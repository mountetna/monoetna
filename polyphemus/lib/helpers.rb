require 'etna/clients'
require 'etna/environment_scoped'
require 'shellwords'

module WithSlackNotifications
  def notify_slack(message, messenger:self.class.name, channel:)
    `/bin/post-to-slack.sh #{Shellwords.escape(messenger)} #{Shellwords.escape(channel)} #{Shellwords.escape(message)} || true`
  end
end

module WithEtnaClients
  def environment
    Polyphemus.instance.environment
  end

  def token
    Polyphemus.instance.config(:polyphemus, environment)[:token]
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
      token: token,
      host: Polyphemus.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
      token: token,
      host: Polyphemus.instance.config(:metis, environment)[:host])
  end

  def metis_client=(client)
    @metis_client = client if Polyphemus.instance.test?
  end
end

module WithLogger
  def logger
    Polyphemus.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
