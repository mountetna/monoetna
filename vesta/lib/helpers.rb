require 'etna/clients'
require 'etna/environment_scoped'
require 'shellwords'

module WithEtnaClients
  def environment
    Vesta.instance.environment
  end

  def token
    Vesta.instance.config(:vesta, environment)[:token]
  end

  def magma_client
    @magma_client ||= Etna::Clients::Magma.new(
      token: token,
      host: Vesta.instance.config(:magma, environment)[:host])
  end

  def metis_client
    @metis_client ||= Etna::Clients::Metis.new(
      token: token,
      host: Vesta.instance.config(:metis, environment)[:host])
  end

  def metis_client=(client)
    @metis_client = client if Vesta.instance.test?
  end

  def janus_client
    @magma_client ||= Etna::Clients::Janus.new(
      token: token,
      host: Vesta.instance.config(:janus, environment)[:host])
  end
end

module WithLogger
  def logger
    Vesta.instance.logger
  end
end

WithEtnaClientsByEnvironment = EnvironmentScoped.new do
  include WithEtnaClients
end
