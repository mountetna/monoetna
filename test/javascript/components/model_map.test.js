import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { mockStore } from '../helpers';
import ModelMap from '../../../lib/client/jsx/components/model_map';

describe('ModelMap', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      location: {
        path: '/labors/browse/monster/Nemean Lion'
      },
      magma: {
        models: {
          patients: {
            template: {
              name: 'patients',
              attributes: {
                link_model_name: 'experiment'
              }
            }
          }
        }
      }
    });
  });

  it('renders', () => {
    global.TIMUR_CONFIG = {
      magma_host: 'magma.test'
    };

    // Wrap with Provider here so store gets passed down to child components in Context
    const tree = renderer
      .create(
        <Provider store={store}>
          <ModelMap />
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
