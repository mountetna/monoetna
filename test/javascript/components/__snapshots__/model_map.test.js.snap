// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelMap renders 1`] = `
<div
  id="map"
>
  <svg
    height={500}
    width={500}
  />
  <div />
</div>
`;
